src/bosh_azure_cpi/lib/cloud/azure/restapi/azure_client.rb

```diff
@@ -383,6 +355,18 @@ def create_virtual_machine(resource_group_name, vm_params, network_interfaces, a
         end
       end
 
+      unless vm_params[:config_disk].nil?
+        # append the config disk into the vm.
+        vm['properties']['storageProfile']['dataDisks'] = [] if vm['properties']['storageProfile']['dataDisks'].nil?
+        vm['properties']['storageProfile']['dataDisks'].push(
+          'lun' => 1,
+          'createOption' => 'Attach',
+          'managedDisk' => {
+            'id' => vm_params[:config_disk][:id].to_s
```

Comment:
config_disk only supports managed disk.
If unmanaged disk is used, where will CPI throw an error?

Reply:
the rest api would fail.
and I will add the unmanaged disk support later.

Reply:
for such cases, it is better to add a TODO comment accordingly.